main.py

```diff
@@ -125,13 +125,18 @@ async def on_typing(channel, user, when):
     logging.debug('Typing from {0.name}'.format(user))
 
 @mbox.event
-async def on_guild_join(guild):
+async def on_guild_join(guild: discord.Guild):
     logging.info('Joined Server: {0.name}'.format(guild))
-    await guild.text_channels[0].send('Thanks for adding Music Bot!')
-    await src.preinitialization.generate_profile(guild, mbox, profiles)
-    for profile in profiles:
-        await profile.setup()
-    print(len(profiles))
+    try:
+        await guild.text_channels[0].send('Thanks for adding Music Bot!')
+        await src.preinitialization.generate_profile(guild, mbox, profiles)
+        for profile in profiles:
+            await profile.setup()
+    except discord.errors.Forbidden:
+        owner_member: discord.Member = guild.owner
+        if owner_member:
+            await owner_member.send(content="You or a member in server ["+ guild.name + "] added me without the right permissions. Try adding me again with the link: " + INVITE_LINK_FORMAT.format(mbox.user.id))
```

Comment:
```suggestion
content= f"You or a member in server  {guild.name} added me without the right permissions. Try adding me again with the link: " + INVITE_LINK_FORMAT.format(mbox.user.id)
await owner_member.send(content=content)
```
Similar to my earlier suggestion, I would recommend using f-strings. Also, I would suggest you to isolate the string by assigning it to its own variable. Just for my understanding, where is INVITE_LINK.format() coming from?

Reply:
Yeah, f-strings are defiantly more concise.

> Just for my understanding, where is INVITE_LINK.format() coming from?
It comes from the new constant variable in constants.py. This is planned to be changed per #46.